### PR TITLE
Support RCTNetworking#clearCookies on iOS

### PR DESCRIPTION
**Pull request by nevir**
This adds cookie clearing support for iOS to match the existing support on Android.  Helpful for resetting the app to a clean state (say, when logging a user out).

### CODE REVIEW

**Comment by facebook-github-bot-3** [comment]
By analyzing the blame information on this pull request, we identified @philikon and @nicklockwood to be potential reviewers.

File: Libraries/Network/RCTNetworking.m [hunk 1]
```diff
@@ -515,6 +515,20 @@ - (RCTNetworkTask *)networkTaskWithRequest:(NSURLRequest *)request
   [_tasksByRequestID removeObjectForKey:requestID];
 }
 
+RCT_EXPORT_METHOD(clearCookies:(RCTResponseSenderBlock)responseSender)
+{
+  NSHTTPCookieStorage* storage = [NSHTTPCookieStorage sharedHTTPCookieStorage];
```
Comment (javache) [inline Libraries/Network/RCTNetworking.m]: nit: our obj-c style is `NSHTTPCookieStorage *storage`

File: Libraries/Network/RCTNetworking.m [hunk 2]
```diff
@@ -515,6 +515,20 @@ - (RCTNetworkTask *)networkTaskWithRequest:(NSURLRequest *)request
...
+  if (!storage.cookies.count) {
+    responseSender(@[@NO]);
+    return;
+  }
+
+  for (NSHTTPCookie* cookie in storage.cookies) {
```
Comment (javache) [inline Libraries/Network/RCTNetworking.m]: same: `NSHTTPCookie *cookie`

Comment (nevir) [inline Libraries/Network/RCTNetworking.m]: Doh, habits :(

**Comment by facebook-github-bot-9** [comment]
@nevir updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9264/files/40f6ada89a3fac06a18be6c7c14217180cc902da..5a7affa35db41494d8c4a9f4be990f53b3dc484e)

**Comment by facebook-github-bot-1** [comment]
@nevir updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9264/files/40f6ada89a3fac06a18be6c7c14217180cc902da..5a7affa35db41494d8c4a9f4be990f53b3dc484e)

**Comment by javache** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-2** [comment]
Thanks for importing.If you are an FB employee go to [Phabricator](https://phabricator.intern.facebook.com/D3776492) to review internal test results.

**Pull request closed without merging**